be/src/io/cache_input_stream.cpp

```diff
@@ -261,9 +280,11 @@ void CacheInputStream::_deduplicate_shared_buffer(SharedBufferedInputStream::Sha
         _block_map.erase(i);
     }
 
-    sb->offset = std::max(start_block_id * _block_size, sb->offset);
-    int64_t end = std::min((end_block_id + 1) * _block_size, end_offset);
-    sb->size = end - sb->offset;
+    if (sb->buffer.capacity() == 0) {
```

Comment:
why add `if` clause here

Reply:
> why add `if` clause here

Because we only change the offset and size of shared buffer to reduce the remote read size. However, when its capacity is not zero, it mean the shared buffer has been filled, now changed the range is meaningless.